lib/yargs-factory.ts

```diff
@@ -2206,6 +2214,15 @@ export class YargsInstance {
   [kSetHasOutput]() {
     this.#hasOutput = true;
   }
+  [kTrackManuallySetKeys](keys: string | string[]) {
+    if (typeof keys === 'string') {
+      this.#options.key[keys] = true;
```

Comment:
doesn't the `this[kPopulateParserHintArray]('boolean', keys);` function also populate `this.#options`?

I'm wondering why we need this additional `kTrackManuallySetKeys`? I'm probably missing something.

Reply:
## Example code

```js
yargs
  .command({
    command: "cmd",
    desc: "cmd desc",
    builder: (yargs) =>
      yargs
        .option("opt1", {
          desc: "opt1 desc",
          type: "boolean",
        })
        .option("opt2", {
          desc: "opt2 desc",
          boolean: true,
        })
        .array("opt3")
        .boolean("opt4")
        .count("opt5")
        .number("opt6")
        .string("opt7")
        .fail(() => {
          console.log('failure!!!')
        }),
    handler: (argv) => {
      console.log(argv);
    },
  })
  .help()
  .strict()
  .parse("cmd --opt1 --opt2 --opt3 foo bar baz --opt4 --opt5 --opt6 3 --opt7 oof");
  ```
## Reason for the change

`yargs.option` has this line:

```js
this.#options.key[keys] = true;
```
This line populates `options.key`. 

Before my fix, calling `yargs.boolean` (or array, count, string, number, etc) was missing this behavior.

To highlight the difference in behavior between `.option()` and `.boolean()`, I added opt1 and opt2 using `.option()`. In the "Without fix" section below, you should see that `options.key` has opt1 and opt2, but is missing the others.

## Explanation 

I logged options (`yargs.getOptions()`) inside of `validation.isValidAndSomeAliasIsNotNew` 

Without the fix, `options.key` is missing keys, so `isValidAndSomeAliasIsNotNew` fails.



### Without fix:

```js
{
  local: [],
  configObjects: [],
  array: [ 'opt3' ],
  boolean: [ 'version', 'help', 'opt1', 'opt2', 'opt4' ],
  string: [ 'opt7' ],
  skipValidation: [],
  count: [ 'opt5' ],
  normalize: [],
  number: [ 'opt6' ],
  hiddenOptions: [],
  narg: {},
  key: { version: true, help: true, opt1: true, opt2: true },
  alias: {},
  default: {},
  defaultDescription: {},
  config: {},
  choices: {},
  demandedOptions: {},
  demandedCommands: {},
  deprecatedOptions: {},
  envPrefix: undefined,
  __: [Function: bound __],
  configuration: {}
}
```

### WIth fix

```js
{
  local: [],
  configObjects: [],
  array: [ 'opt3' ],
  boolean: [ 'version', 'help', 'opt1', 'opt2', 'opt4' ],
  string: [ 'opt7' ],
  skipValidation: [],
  count: [ 'opt5' ],
  normalize: [],
  number: [ 'opt6' ],
  hiddenOptions: [],
  narg: {},
  key: {
    version: true,
    help: true,
    opt1: true,
    opt2: true,
    opt3: true,
    opt4: true,
    opt5: true,
    opt6: true,
    opt7: true
  },
  alias: {},
  default: {},
  defaultDescription: {},
  config: {},
  choices: {},
  demandedOptions: {},
  demandedCommands: {},
  deprecatedOptions: {},
  envPrefix: undefined,
  __: [Function: bound __],
  configuration: {}
}
```